blockchain/chain/src/store/chain_store.rs

```diff
@@ -157,6 +159,22 @@ where
         self.heaviest.read().await.clone()
     }
 
+    pub fn heaviest_tipset_rwlock(&self) -> &RwLock<Option<Arc<Tipset>>> {
+        &self.heaviest
+    }
+
+    pub fn tip_index_rwlock(&self) -> &RwLock<TipIndex> {
+        &self.tip_index
+    }
+
+    pub fn publisher_rwlock(&self) -> &RwLock<Publisher<HeadChange>> {
+        &self.publisher
+    }
+
+    pub fn blockstore_arc(&self) -> Arc<DB> {
```

Comment:
You really shouldn't be describing these functions with the field type. Reason being is that if that type ever changes, you need to either have to rename the function and make breaking changes, or leave and have the name be misleading.

I suggest taking off the `_rwlock` suffixes, and switching the `_arc` suffix to `_cloned`